controllers/spacebindingrequest/spacebindingrequest_controller_test.go

```diff
@@ -0,0 +1,147 @@
+	require.NoError(t, err)
+	sbr := spacebindingrequesttest.NewSpaceBindingRequest("jane", "jane-tenant",
+		spacebindingrequesttest.WithMUR("jane"),
+		spacebindingrequesttest.WithSpaceRole("admin"))
+	t.Run("success", func(t *testing.T) {
+
+		// given
+		member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbr), "member-1", corev1.ConditionTrue)
+		hostClient := test.NewFakeClient(t)
+		ctrl := newReconciler(t, hostClient, member1)
+
+		// when
+		_, err = ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+		// then
+		require.NoError(t, err)
+		// spacebindingrequest exists with expected cluster roles and finalizer
+		spacebindingrequesttest.AssertThatSpaceBindingRequest(t, "jane-tenant", sbr.GetName(), member1.Client).
+			HasSpecSpaceRole("admin").
+			HasSpecMasterUserRecord("jane")
+	})
+
+	t.Run("failure", func(t *testing.T) {
+		t.Run("unable to find SpaceBindingRequest", func(t *testing.T) {
+			// given
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+			// then
+			// space binding request should not be there,
+			// but it should not return an error by design
+			require.NoError(t, err)
+		})
+		t.Run("unable to get SpaceBindingRequest", func(t *testing.T) {
+			member1Client := test.NewFakeClient(t)
+			member1Client.MockGet = mockGetSpaceBindingRequestFail(member1Client)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+			// then
+			// space binding request should not be there
+			require.EqualError(t, err, "unable to get the current SpaceBindingRequest: mock error")
+		})
+
+	})
+}
+
+func newReconciler(t *testing.T, hostCl runtimeclient.Client, memberClusters ...*commoncluster.CachedToolchainCluster) *spacebindingrequest.Reconciler {
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	clusters := map[string]cluster.Cluster{}
+	for _, c := range memberClusters {
+		restClient, err := commontest.NewRESTClient("fake_secret", c.APIEndpoint)
+		restClient.Client.Transport = gock.DefaultTransport // make sure that the underlying client's request are intercepted by Gock
+		require.NoError(t, err)
+		clusters[c.Name] = cluster.Cluster{
+			Config: &commoncluster.Config{
+				Type:              commoncluster.Member,
+				APIEndpoint:       c.APIEndpoint,
+				OperatorNamespace: c.OperatorNamespace,
+				OwnerClusterName:  test.MemberClusterName,
+				RestConfig: &rest.Config{
+					Host: c.APIEndpoint,
+					TLSClientConfig: rest.TLSClientConfig{
+						CAData: []byte("ASDf33=="),
+					},
+				},
+			},
+			RESTClient: restClient,
+			Client:     c.Client,
+		}
```

Comment:
I'm wondering why do actually need this gock mocker, the RESTClient and the RestConfig? We don't need to call any special REST endpoint as we do in SpaceRequest for getting the tokens, right?
wouldn't be this sufficient?
```suggestion
		clusters[c.Name] = cluster.Cluster{
			Config: &commoncluster.Config{
				Type:              commoncluster.Member,
				OperatorNamespace: c.OperatorNamespace,
				OwnerClusterName:  test.MemberClusterName,
			},
			Client: c.Client,
		}
```